File: airbyte-integrations/connectors/destination-e2e-test/src/main/java/io/airbyte/integrations/destination/e2e_test/logging/BaseLogger.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.e2e_test.logging;

import io.airbyte.protocol.models.v0.AirbyteRecordMessage;
import io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair;
import java.time.Instant;
import java.time.OffsetDateTime;
import java.time.ZoneId;
import java.time.format.DateTimeFormatter;

public abstract class BaseLogger implements TestingLogger {

  protected final AirbyteStreamNameNamespacePair streamNamePair;
  protected final int maxEntryCount;
  protected int loggedEntryCount = 0;

  public BaseLogger(final AirbyteStreamNameNamespacePair streamNamePair, final int maxEntryCount) {
    this.streamNamePair = streamNamePair;
    this.maxEntryCount = maxEntryCount;
  }

  protected String entryMessage(final AirbyteRecordMessage recordMessage) {
    return String.format("[%s] %s #%04d: %s",
        emissionTimestamp(recordMessage.getEmittedAt()),
        streamName(streamNamePair),
        loggedEntryCount,
        recordMessage.getData());
  }

  protected static String streamName(final AirbyteStreamNameNamespacePair pair) {
    if (pair.getNamespace() == null) {
      return pair.getName();
    } else {
      return String.format("%s.%s", pair.getNamespace(), pair.getName());
    }
  }

  protected static String emissionTimestamp(final long emittedAt) {
    return OffsetDateTime
        .ofInstant(Instant.ofEpochMilli(emittedAt), ZoneId.systemDefault())
        .format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
  }

}

File: airbyte-integrations/connectors/destination-dev-null/src/main/kotlin/io/airbyte/integrations/destination/dev_null/logging/BaseLogger.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.integrations.destination.dev_null.logging

import io.airbyte.protocol.models.v0.AirbyteRecordMessage
import io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair
import java.time.Instant
import java.time.OffsetDateTime
import java.time.ZoneId
import java.time.format.DateTimeFormatter

abstract class BaseLogger(
    protected val streamNamePair: AirbyteStreamNameNamespacePair,
    protected val maxEntryCount: Int
) : TestingLogger {
    protected var loggedEntryCount: Int = 0

    protected fun entryMessage(recordMessage: AirbyteRecordMessage): String {
        return String.format(
            "[%s] %s #%04d: %s",
            emissionTimestamp(recordMessage.emittedAt),
            streamName(streamNamePair),
            loggedEntryCount,
            recordMessage.data
        )
    }

    companion object {
        protected fun streamName(pair: AirbyteStreamNameNamespacePair): String {
            return if (pair.namespace == null) {
                pair.name
            } else {
                String.format("%s.%s", pair.namespace, pair.name)
            }
        }

        protected fun emissionTimestamp(emittedAt: Long): String {
            return OffsetDateTime.ofInstant(Instant.ofEpochMilli(emittedAt), ZoneId.systemDefault())
                .format(DateTimeFormatter.ISO_OFFSET_DATE_TIME)
        }
    }
}
